### ISSUE TITLE
Converting html to docx fails

### ISSUE THREAD

**Issue by juckerf** [gyselroth/balloon#291]
#### Description
**Is**:
Shadow conversion from html file to docx fails.
```
balloon-jobs-578c6f64c7-9g8qx balloon-jobs Error: no export filter for /tmp/phpCOqbkd.docx found, aborting.
balloon-jobs-578c6f64c7-9g8qx balloon-jobs Error: no export filter
balloon-jobs-578c6f64c7-9g8qx balloon-jobs 2019-27-02 12:54:33 [Balloon\Converter,ERROR]: failed execute adapter [Balloon\Converter\Adapter\Office]  [object] (Balloon\Converter\Exception(code: 0): failed convert document into docx at /usr/share/balloon/src/lib/Converter/Adapter/Office.php:294)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs [stacktrace]
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #0 /usr/share/balloon/src/lib/Converter.php(173): Balloon\Converter\Adapter\Office->convert(Object(Balloon\Filesystem\Node\File), 'docx')
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #1 /usr/share/balloon/src/app/Balloon.App.Convert/Converter.php(177): Balloon\Converter->convert(Object(Balloon\Filesystem\Node\File), 'docx')
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #2 /usr/share/balloon/src/app/Balloon.App.Convert/Job.php(59): Balloon\App\Convert\Converter->convert(Array, Object(Balloon\Filesystem\Node\File))
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #3 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(574): Balloon\App\Convert\Job->start()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #4 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(482): TaskScheduler\Worker->executeJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #5 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(333): TaskScheduler\Worker->processJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #6 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(227): TaskScheduler\Worker->queueJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #7 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(270): TaskScheduler\Worker->processAll()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #8 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(456): TaskScheduler\WorkerManager->spawnWorker()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #9 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(315): TaskScheduler\WorkerManager->handleJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #10 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(176): TaskScheduler\WorkerManager->main()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #11 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Queue.php(148): TaskScheduler\WorkerManager->process()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #12 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Queue.php(93): TaskScheduler\Queue->initWorkerManager()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #13 /usr/share/balloon/src/app/Balloon.App.Cli/Console/Jobs.php(83): TaskScheduler\Queue->process()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #14 /usr/share/balloon/src/lib/Bootstrap/Cli.php(91): Balloon\App\Cli\Console\Jobs->__invoke()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #15 /usr/share/balloon/src/lib/Bootstrap/Cli.php(115): Balloon\Bootstrap\Cli->executeCommand(Object(Balloon\App\Cli\Console\Jobs))
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #16 /usr/share/balloon/src/lib/Bootstrap/Cli.php(76): Balloon\Bootstrap\Cli->routeCommand()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #17 /usr/share/balloon/bin/console/ballooncli(26): Balloon\Bootstrap\Cli->process()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #18 {main}
balloon-jobs-578c6f64c7-9g8qx balloon-jobs 
balloon-jobs-578c6f64c7-9g8qx balloon-jobs 2019-27-02 12:54:33 [TaskScheduler\Worker,ERROR]: failed execute job [5c7688886ed3e20009373b31] of type [Balloon\App\Convert\Job] on worker [5c755b68d8fa00000869cfd7]  [object] (Balloon\Converter\Exception(code: 0): all adapter failed at /usr/share/balloon/src/lib/Converter.php:186)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs [stacktrace]
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #0 /usr/share/balloon/src/app/Balloon.App.Convert/Converter.php(177): Balloon\Converter->convert(Object(Balloon\Filesystem\Node\File), 'docx')
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #1 /usr/share/balloon/src/app/Balloon.App.Convert/Job.php(59): Balloon\App\Convert\Converter->convert(Array, Object(Balloon\Filesystem\Node\File))
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #2 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(574): Balloon\App\Convert\Job->start()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #3 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(482): TaskScheduler\Worker->executeJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #4 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(333): TaskScheduler\Worker->processJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #5 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Worker.php(227): TaskScheduler\Worker->queueJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #6 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(270): TaskScheduler\Worker->processAll()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #7 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(456): TaskScheduler\WorkerManager->spawnWorker()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #8 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(315): TaskScheduler\WorkerManager->handleJob(Array)
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #9 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/WorkerManager.php(176): TaskScheduler\WorkerManager->main()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #10 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Queue.php(148): TaskScheduler\WorkerManager->process()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #11 /usr/share/balloon/vendor/gyselroth/mongodb-php-task-scheduler/src/Queue.php(93): TaskScheduler\Queue->initWorkerManager()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #12 /usr/share/balloon/src/app/Balloon.App.Cli/Console/Jobs.php(83): TaskScheduler\Queue->process()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #13 /usr/share/balloon/src/lib/Bootstrap/Cli.php(91): Balloon\App\Cli\Console\Jobs->__invoke()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #14 /usr/share/balloon/src/lib/Bootstrap/Cli.php(115): Balloon\Bootstrap\Cli->executeCommand(Object(Balloon\App\Cli\Console\Jobs))
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #15 /usr/share/balloon/src/lib/Bootstrap/Cli.php(76): Balloon\Bootstrap\Cli->routeCommand()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #16 /usr/share/balloon/bin/console/ballooncli(26): Balloon\Bootstrap\Cli->process()
balloon-jobs-578c6f64c7-9g8qx balloon-jobs #17 {main}
balloon-jobs-578c6f64c7-9g8qx balloon-jobs 
```

**Should**:
Shadow conversion from html file to docx should work. Probably other format combinations are also affected.

#### balloon server version
tested on 2.4.4 and 2.5.0 branch

#### PHP version
As in docker image gyselroth/balloon-development:latest (PHP 7.2.9)

#### Operating system
As in docker image gyselroth/balloon-development:latest (Debian GNU/Linux 9.5 (stretch))

#### Webserver
As in docker image gyselroth/balloon-development:latest (nginx/1.10.3)

**Comment by juckerf** [comment]
If I specify the conversion filter explicitly in the libreoffice commandline (`--convert-to "docx:MS Word 2007 XML"`) the conversion works.
So I guess balloon should not rely on the automatic filter detection of libreoffice, instead balloon should define the filter to use for every format explicitly (like unoconv does https://github.com/dagwieers/unoconv/blob/master/unoconv#L364)

**Comment by raffis** [comment]
This has been fixed in v2.5 using the new office converter adapter, see #296.
